Index: homeassistant/components/mqtt/update.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/homeassistant/components/mqtt/update.py b/homeassistant/components/mqtt/update.py
--- a/homeassistant/components/mqtt/update.py	(revision e86e79384218e7ec401dce7cc542e27277fcb810)
+++ b/homeassistant/components/mqtt/update.py	(date 1749754817489)
@@ -213,9 +213,6 @@
         if "release_url" in json_payload:
             self._attr_release_url = json_payload["release_url"]

-        if "entity_picture" in json_payload:
-            self._attr_entity_picture = json_payload["entity_picture"]
-
         if "update_percentage" in json_payload:
             self._attr_update_percentage = json_payload["update_percentage"]
             self._attr_in_progress = self._attr_update_percentage is not None
